client/signup/steps/user/index.jsx

```diff
@@ -73,7 +74,17 @@ export class UserStep extends Component {
 	}
 
 	componentWillMount() {
+		const { oauth2Signup, initialContext } = this.props,
+			clientId =
+				initialContext && initialContext.query && initialContext.query.oauth2_client_id
+					? initialContext.query.oauth2_client_id
+					: null;
+
 		this.setSubHeaderText( this.props );
+
+		if ( oauth2Signup && clientId ) {
+			this.props.fetchOAuth2ClientData( clientId );
```

Comment:
Good call fetching the oauth2 client data here I think.

In theory the controller should handle it (so it can customize the layout to load the Oauth2Masterbar for instance like in [`login`](https://github.com/Automattic/wp-calypso/blob/master/client/login/controller.js#L70)). But here in the signup framework I feel like adding more edge cases to [the signup controller](https://github.com/Automattic/wp-calypso/blob/master/client/signup/controller.js) isn't going to help.

Could have been added to the SignupForm as well but I'm not sure it's a better option.